examples/dataflow/chatbot/operations.py

```diff
@@ -0,0 +1,217 @@
+import io
+import re
+import sys
+import json
+import tempfile
+import contextlib
+from aiohttp import ClientSession, ClientTimeout, web
+
+from dffml.cli.cli import CLI
+from dffml import op, config, Definition, BaseSecret
+
+ACCESSTOKEN = Definition(name="access_tok3n", primitive="str")
+ROOMNAME = Definition(name="room_name", primitive="str")
+ROOMID = Definition(name="room_id", primitive="str")
+MESSAGE = Definition(name="message", primitive="str")
+TOSEND = Definition(name="to_send", primitive="str")
+
+
+@config
+class GitterChannelConfig:
+    secret: BaseSecret
+
+
+@op(
+    inputs={"room_uri": ROOMNAME},
+    outputs={"room_id": ROOMID},
+    config_cls=GitterChannelConfig,
+    imp_enter={
+        "secret": lambda self: self.config.secret,
+        "session": lambda self: ClientSession(trust_env=True),
+    },
+    ctx_enter={"sctx": lambda self: self.parent.secret()},
+)
+async def get_room_id(self, room_uri):
+    # Get unique roomid from room uri
+    access_token = await self.sctx.get("access_token")
+    headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "Authorization": f"Bearer {access_token}",
+    }
+
+    data = '{"uri":' + f'"{room_uri}"' + "}"
+    url = "https://api.gitter.im/v1/rooms"
```

Comment:
Let's make the base part of this URL a part of the config,something like `api: str = "https://api.gitter.im/v1"`

We also need to to write a test for this whole thing. We can use the `aiohttp` testing interfaces, or `httptest`